### DISCUSSION TITLE
coinbase wallet not found in mobile

### DISCUSSION THREAD

**Discussion by ryuk8000** [WalletConnect/walletconnect-monorepo#2451] [Developer Support]
why coinbase is not found in mobile even if i have installed it?

import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { Web3Modal } from '@web3modal/html'
import { configureChains, createConfig } from '@wagmi/core'
import { arbitrum, mainnet, polygon } from '@wagmi/core/chains'

const chains = [arbitrum, mainnet, polygon]
const projectId = 'YOUR_PROJECT_ID'

const { publicClient } = configureChains(chains, [w3mProvider({ projectId })])
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({ projectId, version: 1, chains }),
  publicClient
})
const ethereumClient = new EthereumClient(wagmiConfig, chains)
const web3modal = new Web3Modal({ projectId }, ethereumClient)  I am using this as shared in the wallet conenct docs for html

**Comment by glitch-txs** [comment]
Can you try with labs? 
https://lab.web3modal.com/v1ManagedReact

Is it iOS or Android?

**Reply by ryuk8000** [reply]
It is android

**Reply by ryuk8000** [reply]
no coinbase is not shown also in the provided link : https://lab.web3modal.com/v1ManagedReact

**Reply by glitch-txs** [reply]
You should see a "Select Wallet" button and that should show you all wallets you have installed. Just tested on Android with CB and works fine on my end

**Reply by ryuk8000** [reply]
ok . I will look into this. I am also facing issue while scanning qr on desktop through math wallet application in mobile. It cannot scan the qr

**Reply by glitch-txs** [reply]
Is the issue still persistent in the labs link?

**Reply by ryuk8000** [reply]
yes i cant see coinbase wallet in labs link when clicking on select wallet

**Reply by ryuk8000** [reply]
https://www.know-about-blockchain.com/nft
 can u connect wallet through mobile and check if coinbase is showing by connecting to wallet in mobile

**Reply by glitch-txs** [reply]
Just tested, it shows the CB as option on my phone when it takes me to the app and then nothing happens, on the labs website it takes me to CB explorer

**Reply by ryuk8000** [reply]
can u also check whether math wallet qr scan is working or not

**Reply by glitch-txs** [reply]
Have you set the project ID properly? Also I would check the deps version, which version are you using?

**Reply by ryuk8000** [reply]
yes i have set the project id properly. 
i am using these package and version
"@wagmi/core": "^0.10.10",
    "@web3modal/ethereum": "^2.4.1",
    "@web3modal/html": "^2.4.1",

**Reply by glitch-txs** [reply]
Okay, so I've tested Math Wallet in [labs website](https://lab.web3modal.com/v1ManagedReact), the issue I found is, it doesn't work in dark mode, if you set the modal on light mode it will work. @0xAsimetriq

**Reply by ryuk8000** [reply]
cool

**Reply by ryuk8000** [reply]
one last question . I am doing signature verification using wagmiSignmessage . It works fine in desktop but in mobile it connects to dapp then returns back to dapp before signature verification takes place. If we go to the mobile wallet we get screen of signature verification but as it already returns to dapp it looks like we didnt get signature verification screen

**Reply by finessevanes** [reply]
Our lead Web3Modal dev is OOO, but he's been tagged to answer this question when they come back next week. Appreciate your patience!

**Reply by xzilja** [reply]
Connection and signature verification are 2 different events. When you return from the wallet, signature request is sent. I believe you should still get a prompt to go to wallet though if it has deeplinks set up properly.

**Reply by ryuk8000** [reply]
can u tell me how to setup deep links or give me any reference.

**Reply by xzilja** [reply]
You see coinbases extension on desktop, to get it working on mobile a wagmi connector is needed. https://wagmi.sh/react/connectors/coinbaseWallet without it cb won't show up as they need proprietary sdk.

**Reply by ryuk8000** [reply]
import { EthereumClient, w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { Web3Modal } from '@web3modal/html'
import { configureChains, createConfig } from '@wagmi/core'
import { arbitrum, mainnet, polygon } from '@wagmi/core/chains'

const chains = [arbitrum, mainnet, polygon]
const projectId = 'YOUR_PROJECT_ID'

const { publicClient } = configureChains(chains, [w3mProvider({ projectId })])
const wagmiConfig = createConfig({
autoConnect: true,
connectors: w3mConnectors({ projectId, version: 1, chains }),
publicClient
})
const ethereumClient = new EthereumClient(wagmiConfig, chains)

How to integrate coinbase connector in this?

**Reply by finessevanes** [reply]
https://wagmi.sh/react/connectors/coinbaseWallet#configuration
